.buildkite/commands/run-unit-tests.sh

```diff
@@ -31,23 +34,20 @@ if [[ "$TESTS_EXIT_STATUS" -ne 0 ]]; then
 fi
 
 echo "--- 🚦 Report Tests Status"
-path_pattern="*/build/test-results/*/*.xml"
-results_files=()
-while IFS= read -r -d '' file; do
-  results_files+=("$file")
-done < <(find . -path "$path_pattern" -type f -name "*.xml" -print0)
-
-for file in "${results_files[@]}"; do
-  if [[ $BUILDKITE_BRANCH == trunk ]] || [[ $BUILDKITE_BRANCH == release/* ]]; then
-    annotate_test_failures "$file" --slack "build-and-ship"
-  else
-    annotate_test_failures "$file"
-  fi
-done
+results_file="$test_results_dir/merged-test-results.xml"
+
+# Merge JUnit results into a single file (for performance reasons with reporting)
```

Comment:
As a side note and only for my curiosity's sake, I'd be interested to better understand why there's such a huge difference in practice before/after. In particular, I'm curious why it took so long — ≈4mn to do buildkite annotations and ≈9mn to collect artifacts — before the merging and merging them together reduces the time taken by so much.

 - Not only am I interested by this for curiosity's sake to better understand what caused this extra time and potentially act on it (would the root cause suggest that we should maybe optimize our scripts like `annotate_test_failures` in the first place?)
 - But also, this change making those steps now taking only 1s compared to mulitiple minutes previously… leaves me a bit wary and wondering if it's doing anything at all… as opposed to that new 1s time being because those scripts exit early without doing anything (e.g. because we overlooked something in this diff and pass them the wrong path to parse, or the merged file wasn't created as expected for some reason, …)

Reply:
Please feel free to examine the `trunk` commit that this branch is branched from `Unit Test WordPress` job: https://buildkite.com/automattic/wordpress-android/builds/19426#01910379-29a1-4bfe-b44e-f0c6276aad00

> would the root cause suggest that we should maybe optimize our scripts like annotate_test_failures in the first place

I don't see there's anything wrong with `annotate_test_failures`, but I don't think it was ever considered to be used in a loop with 528 iterations (each `annotate_test_failures` AFAIU performs communication with Buildkite API).

> But also, this change making those steps now taking only 1s compared to mulitiple minutes previously… leaves me a bit wary and wondering if it's doing anything at all…

It does - please see `View in Test Analytics` - the number of tests is the same. `Artifacts` contains merged report of all tests, this also works fine.

For annotations part, I'll prepare a temporary unit test failure to make sure that annotations still work.

Reply:
> I don't see there's anything wrong with annotate_test_failures, but I don't think it was ever considered to be used in a loop with 528 iterations (each annotate_test_failures AFAIU performs communication with Buildkite API).

Ooof, 528 iterations each with a Buildkite API call seems a lot, that could indeed explain it all 😅  Thanks for the links 👍

Reply:
> For annotations part, I'll prepare a temporary unit test failure to make sure that annotations still work.

Works as expected! https://buildkite.com/automattic/wordpress-android/builds/19511#019122ab-dbf7-4018-aced-830561a73c21

<img width="1221" alt="image" src="https://github.com/user-attachments/assets/6e591ff5-6435-4a80-b651-947183b64a0c">